src/Module/Photo.php

```diff
@@ -82,24 +82,28 @@ public static function rawContent(array $parameters = [])
 				$square_resize = !in_array($parameters['type'], ['media', 'preview']);
 			}
 
-			if (!empty($parameters['nickname_ext'])) {
-				if (in_array($parameters['type'], ['contact', 'header'])) {
-					$guid = pathinfo($parameters['nickname_ext'], PATHINFO_FILENAME);
-					$account = DBA::selectFirst('account-user-view', ['id'], ['guid' => $guid], ['order' => ['uid' => true]]);
-					if (empty($account)) {
-						throw new HTTPException\NotFoundException();
-					}
+			if (!empty($parameters['guid'])) {
+				$guid = pathinfo($parameters['guid'], PATHINFO_FILENAME);
+				$account = DBA::selectFirst('account-user-view', ['id'], ['guid' => $guid], ['order' => ['uid' => true]]);
+				if (empty($account)) {
+					throw new HTTPException\NotFoundException();
+				}
 
-					$id = $account['id'];
-				} else {
-					$nickname = pathinfo($parameters['nickname_ext'], PATHINFO_FILENAME);
-					$user = User::getByNickname($nickname, ['uid']);
-					if (empty($user)) {
-						throw new HTTPException\NotFoundException();
-					}
+				$id = $account['id'];
+			}
+
+			if (!empty($parameters['contact_id'])) {
+				$id = intval(pathinfo($parameters['contact_id'], PATHINFO_FILENAME));
+			}
```

Comment:
`$parameters['contact_id']` can be 0 for the default avatar, please use `isset()` instead. Additionally, we do not expect any extension in this parameter given the `\d+` pattern we use to match it.
```suggestion
			// Contact Id Fallback, to remove after version 2021.12
			if (!empty($parameters['contact_id'])) {
				$id = intval($parameters['contact_id']);
			}
```